src/lambda_function.py

```diff
@@ -21,17 +22,46 @@ def lambda_handler(event, context):
     if (not validators.url(url)):
         return json.dumps({"error" : "Invalid URL"})
     
-    credibilityresult = cr.getCredibilityScore(url)
-    
+    #### Define the object skeleton
     object = {
         "url" : url,
-        "results" : [
-            credibilityresult
-        ]
+        "article" : {},
+        "results" : []
     }
-    
-    jsonresponse = json.dumps(object)
-    return jsonresponse
 
+    try:
+        credibilityresult = cr.getCredibilityScore(url)
+        object['results'].append(credibilityresult)
+    except Exception as e:
+        object['results'].append({'type': 'credibility', 'outcome': {'score': -1}})
+        
+    try:
+        sentanalysisresult = sa.sentimentAnalysis(url)
+        object['article'] = {'header': sentanalysisresult['header'], 
+                             'summary': sentanalysisresult['summary'],
+                             'keywords': sentanalysisresult['keywords']}
+        object['results'].append({'type': 'polarity',     'outcome': sentanalysisresult['polarity']})
+        object['results'].append({'type': 'subjectivity', 'outcome': sentanalysisresult['subjectivity']})
+    except Exception as e:
+        object['article'] = {'error': 'The article could not be retrieved.'}
+        object['results'].append({'type': 'polarity',     'error': 'no data available'})
```

Comment:
We should standardise how objects/results are returned - it's slightly different between components.